### ISSUE TITLE
Error while running Elixir.Credo.Check.Consistency.SpaceAroundOperators

### ISSUE THREAD

**Issue by KamilLelonek** [rrrene/credo#469]
### Environment

* Credo version (`mix credo -v`): **0.8.8**
* Erlang/Elixir version (`elixir -v`): 

```bash
Erlang/OTP 20 [erts-9.1.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.6.0-dev (4b930b0) (compiled with OTP 20)
```

* Operating system: *MacOS High Sierra*

### What were you trying to do?

    mix credo

### Expected outcome

```bash
Checking 12 source files ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.1 seconds (0.02s to load, 0.1s running checks)
38 mods/funs, found no issues.
```

### Actual outcome

```bash
Checking 12 source files ...
Error while running Elixir.Credo.Check.Consistency.SpaceAroundOperators
** (EXIT from #PID<0.73.0>) an exception was raised:
    ** (MatchError) no match of right hand side value: {:ok, [{:identifier, {1, {1, 10}, nil}, :defmodule}, {:alias, {1, {11, 21}, nil}, :CrmStorage}, {:., {1, {21, 22}, nil}}, {:alias, {1, {22, 33}, nil}, :Application}, {:do, {1, {34, 36}, nil}}, {:eol, {1, {36, 37}, 1}}, {:identifier, {2, {3, 6}, nil}, :use}, {:alias, {2, {7, 18}, nil}, :Application}, {:eol, {2, {18, 19}, 2}}, {:identifier, {4, {3, 8}, nil}, :alias}, {:alias, {4, {9, 22}, nil}, :CrmStorageWeb}, {:., {4, {22, 23}, nil}}, {:alias, {4, {23, 31}, nil}, :Endpoint}, {:eol, {4, {31, 32}, 2}}, {:identifier, {6, {3, 6}, nil}, :def}, {:paren_identifier, {6, {7, 12}, nil}, :start}, {:"(", {6, {12, 13}, nil}}, {:identifier, {6, {13, 18}, nil}, :_type}, {:",", {6, {18, 19}, 0}}, {:identifier, {6, {20, 25}, nil}, :_args}, {:")", {6, {25, 26}, nil}}, {:",", {6, {26, 27}, 1}}, {:kw_identifier, {7, {5, 8}, nil}, :do}, {:alias, {7, {9, 19}, nil}, :Supervisor}, {:., {7, {19, 20}, nil}}, {:paren_identifier, {7, {20, 30}, nil}, :start_link}, {:"(", {7, {30, 31}, nil}}, {:paren_identifier, {7, {31, 39}, nil}, :children}, {:"(", {7, {39, 40}, nil}}, {:")", {7, {40, 41}, nil}}, {:",", {7, {41, 42}, 0}}, {:paren_identifier, {7, {43, 47}, nil}, :opts}, {:"(", {7, {47, 48}, nil}}, {:")", {7, {48, 49}, nil}}, {:")", {7, {49, 50}, nil}}, {:eol, {7, {50, 51}, 2}}, {:identifier, {9, {3, 7}, nil}, :defp}, {:do_identifier, {9, {8, 16}, nil}, :children}, {:do, {9, {17, 19}, nil}}, {:eol, {9, {19, 20}, 1}}, {:"[", {10, {5, 6}, nil}}, {:eol, {10, {6, 7}, 1}}, {:alias, {11, {7, ...}, nil}, :Endpoint}, {:",", {11, {...}, ...}}, {:"]", {12, ...}}, {:eol, {...}}, {:end, ...}, {...}, ...]}
        lib/credo/code.ex:104: Credo.Code.to_tokens/1
        lib/credo/check/consistency/space_around_operators/collector.ex:10: Credo.Check.Consistency.SpaceAroundOperators.Collector.collect_matches/2
        lib/credo/check/consistency/collector.ex:126: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
        (elixir) lib/enum.ex:1289: Enum."-map/2-lists^map/1-0-"/2
        lib/credo/check/consistency/collector.ex:125: Credo.Check.Consistency.Collector.find_issues/4
        lib/credo/check/consistency/space_around_operators/collector.ex:2: Credo.Check.Consistency.SpaceAroundOperators.Collector.find_and_append_issues/4
        lib/credo/check/runner.ex:148: Credo.Check.Runner.run_check/3
        (elixir) lib/task/supervised.ex:88: Task.Supervised.do_apply/2
```

It does work on `Elixir 1.5.2` though.

**Comment by rdrey** [comment]
looks like a duplicate of https://github.com/rrrene/credo/issues/463

**Comment by rrrene** [comment]
@KamilLelonek Thx for your bug report! The initial problem is now fixed on `master`.

I will close this as duplicate of #463 in favor of the discussion in that issue.

Thanks again! :+1:

**Comment by cdesch** [comment]
I'm getting this error on Credo `0.9.0-rc3` with elixir `1.6.1`

```
cjs-MBP15:faker cj$  mix credo --strict 
Checking 50 source files ...
Error while running Elixir.Credo.Check.Consistency.SpaceInParentheses
** (EXIT from #PID<0.73.0>) an exception was raised:
    ** (Protocol.UndefinedError) protocol String.Chars not implemented for {:sigil, {33, 44, nil}, 115, ["  "], [], "("}
        (elixir) lib/string/chars.ex:3: String.Chars.impl_for!/1
        (elixir) lib/string/chars.ex:22: String.Chars.to_string/1
        lib/credo/code/token.ex:243: Credo.Code.Token.Elixir1_6_0.to_col_end/3
        lib/credo/code/token.ex:237: Credo.Code.Token.Elixir1_6_0.convert_to_col_end/3
        (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
        lib/credo/code/token.ex:130: Credo.Code.Token.Elixir1_6_0.position_tuple_for_quoted_string/3
        lib/credo/code/token.ex:85: Credo.Code.Token.Elixir1_6_0.position/1
        lib/credo/code/interpolation_helper.ex:157: Credo.Code.InterpolationHelper.find_interpolations/2


00:12:16.153 [error] Task #PID<0.192.0> started from #PID<0.73.0> terminating
** (Protocol.UndefinedError) protocol String.Chars not implemented for {:sigil, {33, 44, nil}, 115, ["  "], [], "("}
    (elixir) lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir) lib/string/chars.ex:22: String.Chars.to_string/1
    lib/credo/code/token.ex:243: Credo.Code.Token.Elixir1_6_0.to_col_end/3
    lib/credo/code/token.ex:237: Credo.Code.Token.Elixir1_6_0.convert_to_col_end/3
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/credo/code/token.ex:130: Credo.Code.Token.Elixir1_6_0.position_tuple_for_quoted_string/3
    lib/credo/code/token.ex:85: Credo.Code.Token.Elixir1_6_0.position/1
    lib/credo/code/interpolation_helper.ex:157: Credo.Code.InterpolationHelper.find_interpolations/2
Function: #Function<7.109245546/0 in Credo.Check.Runner.run_checks_that_run_on_all/2>
    Args: []

00:12:16.158 [error] GenServer Credo.Supervisor terminating
** (Protocol.UndefinedError) protocol String.Chars not implemented for {:sigil, {33, 44, nil}, 115, ["  "], [], "("}
    (elixir) lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir) lib/string/chars.ex:22: String.Chars.to_string/1
    lib/credo/code/token.ex:243: Credo.Code.Token.Elixir1_6_0.to_col_end/3
    lib/credo/code/token.ex:237: Credo.Code.Token.Elixir1_6_0.convert_to_col_end/3
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/credo/code/token.ex:130: Credo.Code.Token.Elixir1_6_0.position_tuple_for_quoted_string/3
    lib/credo/code/token.ex:85: Credo.Code.Token.Elixir1_6_0.position/1
    lib/credo/code/interpolation_helper.ex:157: Credo.Code.InterpolationHelper.find_interpolations/2
Last message: {:EXIT, #PID<0.73.0>, {%Protocol.UndefinedError{description: "", protocol: String.Chars, value: {:sigil, {33, 44, nil}, 115, ["  "], [], "("}}, [{String.Chars, :impl_for!, 1, [file: 'lib/string/chars.ex', line: 3]}, {String.Chars, :to_string, 1, [file: 'lib/string/chars.ex', line: 22]}, {Credo.Code.Token.Elixir1_6_0, :to_col_end, 3, [file: 'lib/credo/code/token.ex', line: 243]}, {Credo.Code.Token.Elixir1_6_0, :convert_to_col_end, 3, [file: 'lib/credo/code/token.ex', line: 237]}, {Enum, :"-reduce/3-lists^foldl/2-0-", 3, [file: 'lib/enum.ex', line: 1899]}, {Credo.Code.Token.Elixir1_6_0, :position_tuple_for_quoted_string, 3, [file: 'lib/credo/code/token.ex', line: 130]}, {Credo.Code.Token.Elixir1_6_0, :position, 1, [file: 'lib/credo/code/token.ex', line: 85]}, {Credo.Code.InterpolationHelper, :find_interpolations, 2, [file: 'lib/credo/code/interpolation_helper.ex', line: 157]}]}}
State: {:state, {:local, Credo.Supervisor}, :one_for_one, [{:child, #PID<0.135.0>, Credo.Service.SourceFileSource, {Credo.Service.SourceFileSource, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileSource]}, {:child, #PID<0.134.0>, Credo.Service.SourceFileLines, {Credo.Service.SourceFileLines, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileLines]}, {:child, #PID<0.133.0>, Credo.Service.SourceFileAST, {Credo.Service.SourceFileAST, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileAST]}, {:child, #PID<0.132.0>, Credo.Service.SourceFileScopes, {Credo.Service.SourceFileScopes, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileScopes]}, {:child, #PID<0.131.0>, Credo.Service.Commands, {Credo.Service.Commands, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.Commands]}, {:child, #PID<0.130.0>, Credo.CLI.Output.Shell, {Credo.CLI.Output.Shell, :start_link, []}, :permanent, 5000, :worker, [Credo.CLI.Output.Shell]}], :undefined, 3, 5, [], 0, Supervisor.Default, {:ok, {%{intensity: 3, period: 5, strategy: :one_for_one}, [{Credo.CLI.Output.Shell, {Credo.CLI.Output.Shell, :start_link, []}, :permanent, 5000, :worker, [Credo.CLI.Output.Shell]}, {Credo.Service.Commands, {Credo.Service.Commands, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.Commands]}, {Credo.Service.SourceFileScopes, {Credo.Service.SourceFileScopes, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileScopes]}, {Credo.Service.SourceFileAST, {Credo.Service.SourceFileAST, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileAST]}, {Credo.Service.SourceFileLines, {Credo.Service.SourceFileLines, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileLines]}, {Credo.Service.SourceFileSource, {Credo.Service.SourceFileSource, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileSource]}]}}}

```

**Comment by emadb** [comment]
Hi all.
Is it possible that Elixir 1.11.x cause this old error.
I get this after upgrading:
```
Checking 87 source files (this might take a while) ...
** (config) Error while running Elixir.Credo.Check.Consistency.SpaceAroundOperators

11:51:36.459 [error] Task #PID<0.323.0> started from #PID<0.94.0> terminating
** (FunctionClauseError) no function clause matching in Credo.Code.Token.PositionHelper.position/1
    lib/credo/code/token.ex:49: Credo.Code.Token.PositionHelper.position({:sigil, {65, 20, nil}, 85, ["2020-04-09 11:00:00.000000Z"], [], nil, "["})
    lib/credo/check/consistency/space_around_operators/space_helper.ex:58: Credo.Check.Consistency.SpaceAroundOperators.SpaceHelper.no_space_between?/2
    lib/credo/check/consistency/space_around_operators/collector.ex:166: Credo.Check.Consistency.SpaceAroundOperators.Collector.without_space?/3
    lib/credo/check/consistency/space_around_operators/collector.ex:130: Credo.Check.Consistency.SpaceAroundOperators.Collector.record_spaces/4
    lib/credo/check/consistency/space_around_operators/collector.ex:37: Credo.Check.Consistency.SpaceAroundOperators.Collector.traverse_tokens/3
    lib/credo/check/consistency/collector.ex:140: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
    (elixir 1.11.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.11.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
Function: #Function<10.46020751/0 in Credo.Check.Runner.run/2>
    Args: []
** (EXIT from #PID<0.94.0>) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Credo.Code.Token.PositionHelper.position/1
        lib/credo/code/token.ex:49: Credo.Code.Token.PositionHelper.position({:sigil, {65, 20, nil}, 85, ["2020-04-09 11:00:00.000000Z"], [], nil, "["})
        lib/credo/check/consistency/space_around_operators/space_helper.ex:58: Credo.Check.Consistency.SpaceAroundOperators.SpaceHelper.no_space_between?/2
        lib/credo/check/consistency/space_around_operators/collector.ex:166: Credo.Check.Consistency.SpaceAroundOperators.Collector.without_space?/3
        lib/credo/check/consistency/space_around_operators/collector.ex:130: Credo.Check.Consistency.SpaceAroundOperators.Collector.record_spaces/4
        lib/credo/check/consistency/space_around_operators/collector.ex:37: Credo.Check.Consistency.SpaceAroundOperators.Collector.traverse_tokens/3
        lib/credo/check/consistency/collector.ex:140: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
        (elixir 1.11.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
        (elixir 1.11.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
```

**Comment by rrrene** [comment]
Could you post the output of `mix credo info`?

**Comment by emadb** [comment]
Mix credo info:
```
System:
  Credo: 1.4.0
  Elixir: 1.11.0
  Erlang: 23
```

Here is the `.credo.exs`: 
```
# This file contains the configuration for Credo and you are probably reading
# this after creating it with `mix credo.gen.config`.
#
# If you find anything wrong or unclear in this file, please report an
# issue on GitHub: https://github.com/rrrene/credo/issues
#
%{
  #
  # You can have as many configs as you like in the `configs:` field.
  configs: [
    %{
      #
      # Run any exec using `mix credo -C <name>`. If no exec name is given
      # "default" is used.
      #
      name: "pick_in_car",
      #
      # These are the files included in the analysis:
      files: %{
        #
        # You can give explicit globs or simply directories.
        # In the latter case `**/*.{ex,exs}` will be used.
        #
        included: ["lib/", "src/", "test/", "web/", "apps/"],
        excluded: [~r"/_build/", ~r"/deps/", ~r"/node_modules/"]
      },
      #
      # If you create your own checks, you must specify the source files for
      # them here, so they can be loaded by Credo before running the analysis.
      #
      requires: [],
      #
      # If you want to enforce a style guide and need a more traditional linting
      # experience, you can change `strict` to `true` below:
      #
      strict: false,
      #
      # If you want to use uncolored output by default, you can change `color`
      # to `false` below:
      #
      color: true,
      #
      # You can customize the parameters of any check by adding a second element
      # to the tuple.
      #
      # To disable a check put `false` as second element:
      #
      #     {Credo.Check.Design.DuplicatedCode, false}
      #
      checks: [
        #
        ## Consistency Checks
        #
        {Credo.Check.Consistency.ExceptionNames, []},
        {Credo.Check.Consistency.LineEndings, []},
        {Credo.Check.Consistency.ParameterPatternMatching, []},
        {Credo.Check.Consistency.SpaceAroundOperators, []},
        {Credo.Check.Consistency.SpaceInParentheses, []},
        {Credo.Check.Consistency.TabsOrSpaces, []},

        #
        ## Design Checks
        #
        # You can customize the priority of any check
        # Priority values are: `low, normal, high, higher`
        #
        {Credo.Check.Design.AliasUsage,
         [priority: :low, if_nested_deeper_than: 2, if_called_more_often_than: 0]},
        # You can also customize the exit_status of each check.
        # If you don't want TODO comments to cause `mix credo` to fail, just
        # set this value to 0 (zero).
        #
        # {Credo.Check.Design.TagTODO, [exit_status: 2]},
        {Credo.Check.Design.TagFIXME, []},

        #
        ## Readability Checks
        #
        {Credo.Check.Readability.AliasOrder, []},
        {Credo.Check.Readability.FunctionNames, []},
        {Credo.Check.Readability.LargeNumbers, []},
        {Credo.Check.Readability.MaxLineLength, [priority: :low, max_length: 120]},
        {Credo.Check.Readability.ModuleAttributeNames, []},
        # {Credo.Check.Readability.ModuleDoc, []},
        {Credo.Check.Readability.ModuleNames, []},
        {Credo.Check.Readability.ParenthesesInCondition, []},
        {Credo.Check.Readability.ParenthesesOnZeroArityDefs, []},
        {Credo.Check.Readability.PredicateFunctionNames, []},
        {Credo.Check.Readability.PreferImplicitTry, []},
        {Credo.Check.Readability.RedundantBlankLines, []},
        {Credo.Check.Readability.Semicolons, []},
        {Credo.Check.Readability.SpaceAfterCommas, []},
        {Credo.Check.Readability.StringSigils, []},
        {Credo.Check.Readability.TrailingBlankLine, []},
        {Credo.Check.Readability.TrailingWhiteSpace, []},
        # TODO: enable by default in Credo 1.1
        {Credo.Check.Readability.UnnecessaryAliasExpansion, false},
        {Credo.Check.Readability.VariableNames, []},

        #
        ## Refactoring Opportunities
        #
        {Credo.Check.Refactor.CondStatements, []},
        {Credo.Check.Refactor.CyclomaticComplexity, []},
        {Credo.Check.Refactor.FunctionArity, []},
        {Credo.Check.Refactor.LongQuoteBlocks, []},
        # {Credo.Check.Refactor.MapInto, []},
        {Credo.Check.Refactor.MatchInCondition, []},
        {Credo.Check.Refactor.NegatedConditionsInUnless, []},
        {Credo.Check.Refactor.NegatedConditionsWithElse, []},
        {Credo.Check.Refactor.Nesting, []},
        # {Credo.Check.Refactor.PipeChainStart,
        #  [
        #    excluded_argument_types: [:atom, :binary, :fn, :keyword, :number],
        #    excluded_functions: []
        #  ]},
        {Credo.Check.Refactor.UnlessWithElse, []},

        #
        ## Warnings
        #
        {Credo.Check.Warning.BoolOperationOnSameValues, []},
        {Credo.Check.Warning.ExpensiveEmptyEnumCheck, []},
        {Credo.Check.Warning.IExPry, []},
        {Credo.Check.Warning.IoInspect, []},
        # {Credo.Check.Warning.LazyLogging, []},
        {Credo.Check.Warning.OperationOnSameValues, []},
        {Credo.Check.Warning.OperationWithConstantResult, []},
        {Credo.Check.Warning.RaiseInsideRescue, []},
        {Credo.Check.Warning.UnusedEnumOperation, []},
        {Credo.Check.Warning.UnusedFileOperation, []},
        {Credo.Check.Warning.UnusedKeywordOperation, []},
        {Credo.Check.Warning.UnusedListOperation, []},
        {Credo.Check.Warning.UnusedPathOperation, []},
        {Credo.Check.Warning.UnusedRegexOperation, []},
        {Credo.Check.Warning.UnusedStringOperation, []},
        {Credo.Check.Warning.UnusedTupleOperation, []},

        #
        # Controversial and experimental checks (opt-in, just replace `false` with `[]`)
        #
        {Credo.Check.Consistency.MultiAliasImportRequireUse, false},
        {Credo.Check.Design.DuplicatedCode, false},
        {Credo.Check.Readability.MultiAlias, false},
        {Credo.Check.Readability.Specs, false},
        {Credo.Check.Refactor.ABCSize, false},
        {Credo.Check.Refactor.AppendSingleItem, false},
        {Credo.Check.Refactor.DoubleBooleanNegation, false},
        {Credo.Check.Refactor.ModuleDependencies, false},
        {Credo.Check.Refactor.VariableRebinding, false},
        {Credo.Check.Warning.MapGetUnsafePass, false},
        {Credo.Check.Warning.UnsafeToAtom, false}

        #
        # Custom checks can be created using `mix credo.gen.check`.
        #
      ]
    }
  ]
}
```

**Comment by rrrene** [comment]
@emadb Could you try upgrading to Credo `1.4.1`?

**Comment by emadb** [comment]
It worked!

**Comment by rrrene** [comment]
No problem :+1:
